### ISSUE TITLE
Fails to parse empty strings

### ISSUE THREAD

**Issue by adrianmay** [HaXml/HaXml#12]
I have a schema saying
```
<?xml version="1.0" encoding="utf-8"?>
<xsd:schema attributeFormDefault="unqualified" elementFormDefault="qualified" xmlns:xsd="http://www.w3.org/2001/XMLSchema">

  <xsd:element name="Bug" type="BugType" />

  <xsd:complexType name="BugType">
    <xsd:sequence>
      <xsd:element minOccurs="1" maxOccurs="1" name="status" type="xs:string"/>
    </xsd:sequence>
  </xsd:complexType>

</xsd:schema>
```
which succeeds on: 
```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?><BugType><status>Foo</status></BugType>
```
but fails on:
```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?><BugType><status></status></BugType>
```

I think the empty string should also work

**Comment by juhp** [comment]
Can you try #10, please?

**Comment by adrianmay** [comment]
The fix works. Please can I have a patch number bump so I can make stuff use it?

**Comment by juhp** [comment]
I released [HaXml-1.25.10](https://hackage.haskell.org/package/HaXml-1.25.10)
